src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheCrusader/instance_trial_of_the_crusader.cpp

```diff
@@ -59,6 +59,40 @@ class instance_trial_of_the_crusader : public InstanceMapScript
         uint64 GO_WebDoorGUID;
         uint64 GO_FloorGUID;
 
+        void SpawnAnubArak()
+        {
+		    if (InstanceProgress == INSTANCE_PROGRESS_ANUB_ARAK)
+		    {
+			    if (Creature* barrett = instance->GetCreature(NPC_BarrettGUID))
+			    {
+				    barrett->SetVisible(false);
+				    if (AttemptsLeft)
+					    if (!ObjectAccessor::GetCreature(*barrett, NPC_AnubarakGUID))
+						    barrett->SummonCreature(NPC_ANUBARAK, Locs[LOC_ANUB].GetPositionX(), Locs[LOC_ANUB].GetPositionY(), Locs[LOC_ANUB].GetPositionZ(), Locs[LOC_ANUB].GetOrientation(), TEMPSUMMON_CORPSE_TIMED_DESPAWN, 630000000);
+			    }
+
+			    // move corpses
+			    static const uint64 Npcs[4] = { NPC_IcehowlGUID, NPC_JaraxxusGUID, NPC_LightbaneGUID, NPC_DarkbaneGUID };
+			    for (auto& i : Npcs)
+			    {
+				    if (Creature* c = instance->GetCreature(i))
+				    {
+					    if (c->GetGUID() == NPC_IcehowlGUID)
```

Comment:
```
switch (c->GetGUID())
{
    case NPC_IcehowlGUID:
        c->UpdatePosition(626.57f, 162.8f, 140.25f, 4.44f, true);
        c->StopMovingOnCurrentPos();
        c->DestroyForNearbyPlayers();
        break;
    case NPC_JaraxxusGUID:
        c->UpdatePosition(603.92f, 102.61f, 141.85f, 1.4f, true);
        c->StopMovingOnCurrentPos();
        c->DestroyForNearbyPlayers();
        break;
    case NPC_LightbaneGUID:
        c->UpdatePosition(634.58f, 147.16f, 140.5f, 3.02f, true);
        c->StopMovingOnCurrentPos();
        c->DestroyForNearbyPlayers();
        break;
    case NPC_DarkbaneGUID:
        c->UpdatePosition(630.88f, 131.39f, 140.8f, 3.02f, true);
        c->StopMovingOnCurrentPos();
        c->DestroyForNearbyPlayers();
        break;
}
``` 
maeby? instead of elsfifing

Reply:
I already tried to use a switch statement but was getting the error:

Error: 'this' cannot be used in a constant expression

Reply:
ahh i see :C

Reply:
Try switch, it should work...

Reply:
if you run in errors you can also try to put { } brackets for each case

Reply:
Kito. https://i.imgur.com/OSp00VH.png

I think we can't use a switch here because the bosses GUIDs are not known at compile time.

> The case statements require integral value which must be known at compile-time

And if I make them static const or static constexpr. They would be usable in a switch but we won't be able to set the correct guid OnCreatureCreate.

If you have any workarounds let me know